src/host-config.ts

```diff
@@ -134,9 +146,19 @@ async function setProxy(maybeConf: ProxyConfig | null): Promise<void> {
 		if (_.isArray(conf.noProxy)) {
 			await writeFileAtomic(noProxyPath, conf.noProxy.join('\n'));
 		}
-		const redsocksConf = `${redsocksHeader}${generateRedsocksConfEntries(
-			conf,
-		)}${redsocksFooter}`;
+
+		let currentConf: ProxyConfig | undefined;
+		try {
+			currentConf = await readProxy();
```

Comment:
I thought we were going to remove this new read? If this fails `currentConf` will be undefined which will make the object spread fail on line 159.

Reply:
Sorry, don't remember discussing this, but you're right, it's an oversight on my part.

Reply:
@pipex Actually, I did a lil sandbox test with undefined spread in TS, and learned that [a falsey spread will be skipped over in TS](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#conditional-spreads-create-optional-properties)

So even if `currentConf` is undefined, `generateRedsocksConfEntries({ ...currentConf, ...conf })` on line 159 below will not error & will simply skip over currentConf.

Reply:
[Here](https://repl.it/@cywang117/OverdueWelcomeBytecode#index.ts) is my sandbox for reference. You can run `node index.js` in the integrated shell to see what's happening.

Thanks for this by the way! I too thought it would error after looking at it again, but as a result learned about this cool feature of TypeScript.

Reply:
And apparently this is just a JS feature! https://www.flowdock.com/app/rulemotion/r-beginners/threads/f_bsL91RV1Zx6FgcQ8-IAuYgWh8

Reply:
I saw that thread, TIL for me as well. This looks good, I think I was confusing this change with something else.